calculator-app/src/Calculator.js

```diff
@@ -0,0 +1,219 @@
+  state = {
+    previousDisplay: "",
+    operation: "",
+    display: "0",
+    sign: "positive",
+    history: "",
+  };
+
+  handleDigit = (userInput) => {
+    const { display, operation } = this.state;
+    if (operation === "=") {
+      this.setState({
+        history: "",
+      });
+    } else if (
+      display === "0") {
+        this.setState({
+            display: userInput,
+            history: userInput,
+          })
+      } else if ( operation === "" ) {
+        this.setState((prevState) => ({
+            display: prevState.display + userInput,
+            history: prevState.history + userInput,
+          }));
+      } else if (operation !== "") {
+        this.setState((prevState) => ({
+          display: "" + userInput,
+          history: prevState.history + userInput,
+        }));
+      }
+  };
+
+  handleOperand = (userInput) => {
+    const { operation, display } = this.state;
+    if (operation !== "") {
+      this.setState({ operation: userInput });
+    } else {
+      this.setState((prevState) => ({
+        operation: userInput,
+        history: prevState.display + userInput,
+        previousDisplay: display,
+      }));
+    }
+  };
+
+  handleEqual = () => {
+    const { previousDisplay, display, operation } = this.state;
+    // add check for commas(,)
+    this.setState((prevState) => ({
+      previousDisplay: "",
+      history: prevState.history + "=",
+      operation: "=",
+    }));
+    switch (operation) {
+      case "+":
+        this.setState({
+          display: Number(previousDisplay) + Number(display),
+        });
+        break;
+      case "-":
+        this.setState({
+          display: Number(previousDisplay) - Number(display),
+        });
+        break;
+      case "x":
+        this.setState({
+          display: Number(previousDisplay) * Number(display),
+        });
+        break;
+      case "÷":
+        this.setState({
+          display: Number(previousDisplay) / Number(display),
+        });
+        break;
+      default:
+        break;
+    }
+  };
+
+  handleDecimal = () => {
+    const { previousDisplay } = this.state;
+    if (!previousDisplay.includes(".")) {
+      this.setState((prevState) => ({
+        previousDisplay: prevState.previousDisplay + ".",
+        display: prevState.display + ".",
+        history: prevState.history + ".",
+      }));
+    }
+  };
+
+  handleDelete = () => {
+    const { previousDisplay, display, history } = this.state;
+    if (previousDisplay === "") {
+      this.setState({
+        previousDisplay: "",
+      });
+    } else {
+      this.setState({
+        previousDisplay: previousDisplay.slice(0, -1),
+        display: display.slice(0, -1),
+        history: history.slice(0, -1),
+      });
+    }
+  };
+
+  handleAllClear = () => {
+    this.setState({
+      previousDisplay: "",
+      display: "0",
+      history: "",
+      operation: "",
+      sign: "positive",
+    });
+  };
+
+  handleSign = () => {
```

Comment:
I think this entire function can be replaced with checking if the sign state is negative and just removing or adding a "-" accordingly